src/deepsparse/transformers/engines/nl_decoder_engine.py

```diff
@@ -155,18 +155,17 @@ def run(self, inputs: List[numpy.ndarray], val_inp: bool) -> List[numpy.ndarray]
         If the self.internal_cache_active=True, the internal
         deepsparse kv cache management is enabled. In this case
         the LIB.kv_cache class object will be passed to the engine
-        call as well.
+        call as well. In this scenario also the inputs will not be
+        validated, even if the val_inp=True. This is because we
+        want to pass the empty kv cache inputs (batch_size=0) to
+        the engine.
 
         :param inputs: The inputs to run the engine with
         :param val_inp: Whether the input is for validation or not
         :return: The output of the engine
         """
 
         if self.internal_cache_active:
-            # validate the inputs if needed
-            if val_inp:
-                self.engine._validate_inputs(inputs)
```

Comment:
I think its worth keeping this check and just printing a warning to the user that their request for input validation is going to be ignored